components/cursor/cursor.hpp

```diff
@@ -40,17 +82,27 @@ namespace components::cursor {
         data_ptr next();
         data_ptr get() const;
         data_ptr get(std::size_t index) const;
+        bool is_success() const noexcept;
+        bool is_error() const noexcept;
+        error_t get_error() const;
         void sort(std::function<bool(data_ptr, data_ptr)> sorter);
 
     private:
         std::size_t size_{};
         index_t current_index_{start_index};
         std::pmr::vector<std::unique_ptr<sub_cursor_t>> sub_cursor_;
         std::pmr::vector<data_ptr> sorted_;
+        error_t error_;
+        bool success_{true};
 
         void create_list_by_sort();
         data_ptr get_sorted(std::size_t index) const;
         data_ptr get_unsorted(std::size_t index) const;
     };
 
+    using cursor_t_ptr = boost::intrusive_ptr<cursor_t>;
+
+    cursor_t_ptr make_cursor(std::pmr::memory_resource* resource, operation_status_t op_status);
+    cursor_t_ptr make_cursor(std::pmr::memory_resource* resource);
+    cursor_t_ptr make_cursor(std::pmr::memory_resource* resource, error_code_t type, const std::string& what = std::string());
```

Comment:
@OrangeSmokingJacket @kotbegemot this function designed only for error cursor, so might be we can rename it into 
`make_error_cursor`
or
`make_error`
in this case much easier to read code when we return an error.